types/events.go

```diff
@@ -39,6 +44,97 @@ func (em EventManager) ABCIEvents() []abci.Event {
 	return em.events.ToABCIEvents()
 }
 
+// EmitTypedEvent takes typed event and emits converting it into Event
+func (em *EventManager) EmitTypedEvent(tev proto.Message) error {
+	event, err := TypedEventToEvent(tev)
+	if err != nil {
+		return err
+	}
+
+	em.EmitEvent(event)
+	return nil
+}
+
+// EmitTypedEvents takes series of typed events and emit
+func (em *EventManager) EmitTypedEvents(tevs ...proto.Message) error {
+	events := make(Events, len(tevs))
+	for i, tev := range tevs {
+		res, err := TypedEventToEvent(tev)
+		if err != nil {
+			return err
+		}
+		events[i] = res
+	}
+
+	em.EmitEvents(events)
+	return nil
+}
+
+// TypedEventToEvent takes typed event and converts to Event object
+func TypedEventToEvent(tev proto.Message) (Event, error) {
+	evtType := proto.MessageName(tev)
+	evtJSON, err := codec.ProtoMarshalJSON(tev, nil)
```

Comment:
I think that the `nil` argument won't work if `tev` contains an `Any`, or maybe it will. Just to be sure, would you mind adding a test with `Any` in `TestEventManagerTypedEvents`?